### PR TITLE
[TM-1237] add filters in gallery secction

### PR DESCRIPTION
**Pull request by dottyy**
[Task link](https://gfw.atlassian.net/jira/software/c/projects/TM/boards/40?selectedIssue=TM-1237)

### CODE REVIEW

File: src/components/elements/ImageGallery/ImageGallery.tsx [hunk 1]
```diff
@@ -3,17 +3,19 @@ import classNames from "classnames";
...
 import Modal from "@/components/extensive/Modal/Modal";
 import { ModalId } from "@/components/extensive/Modal/ModalConst";
 import Pagination from "@/components/extensive/Pagination";
 import { VARIANT_PAGINATION_TEXT_16 } from "@/components/extensive/Pagination/PaginationVariant";
 import { useModalContext } from "@/context/modal.provider";
 import { Option } from "@/types/common";
 
+import Menu, { MenuItemProps } from "../Menu/Menu";
+import MenuColapse from "../Menu/MenuCollapse";
+import { MENU_PLACEMENT_BOTTOM_BOTTOM } from "../Menu/MenuVariant";
```
Comment (pachonjcl) [inline src/components/elements/ImageGallery/ImageGallery.tsx:18]: imports should use non relative paths please

File: src/components/elements/Menu/MenuCollapse.tsx [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+import { ReactNode, useEffect, useRef, useState } from "react";
+import React from "react";
+import { Else, If, Then } from "react-if";
+import { twMerge as tw } from "tailwind-merge";
+
+import Icon, { IconNames } from "@/components/extensive/Icon/Icon";
+
+import { MenuItem } from "../MenuItem/MenuItem";
+import { MENU_ITEM_VARIANT_BLUE } from "../MenuItem/MenuItemVariant";
+import { MenuItemProps } from "./Menu";
```
Comment (pachonjcl) [inline src/components/elements/Menu/MenuCollapse.tsx:11]: same here

File: src/components/elements/Menu/MenuCollapse.tsx [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    const handleClickOutside = (event: MouseEvent) => {
+      if (menuContainerRef.current && !menuContainerRef.current?.contains(event.target as Node)) {
+        setIsOpen(false);
+      }
+    };
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
```
Comment (pachonjcl) [inline src/components/elements/Menu/MenuCollapse.tsx:76]: don't we have it as a reusable hook? if not, can you create it?

File: src/components/elements/Menu/MenuCollapse.tsx [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    };
+    container?.addEventListener("scroll", hideMenu);
+    window.addEventListener("scroll", hideMenu);
+    hideMenu();
+
+    return () => {
+      container?.removeEventListener("scroll", hideMenu);
+      window.removeEventListener("scroll", hideMenu);
+    };
+  }, [container]);
```
Comment (pachonjcl) [inline src/components/elements/Menu/MenuCollapse.tsx:95]: same here, good candidate for a reusable hook

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: src/components/elements/ImageGallery/ImageGallery.tsx (hunk 1 above)
Comment (dottyy) [reply src/components/elements/ImageGallery/ImageGallery.tsx]: done

File: src/components/elements/Menu/MenuCollapse.tsx (hunk 4 above)
Comment (dottyy) [reply src/components/elements/Menu/MenuCollapse.tsx]: done

File: src/components/elements/Menu/MenuCollapse.tsx (hunk 3 above)
Comment (dottyy) [reply src/components/elements/Menu/MenuCollapse.tsx]: done

**Review by pachonjcl — APPROVED** [review]

**Pull request merged**